service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -181,6 +187,39 @@ private DatasetPreviewTable generateDatarepoTable(
                 .getResult());
   }
 
+  private DatasetPreviewTable generateRawlsTable(
+      AuthenticatedUserRequest user, Dataset dataset, String tableName) {
+    Map<String, EntityTypeMetadata> ent =
+        rawlsService.entityMetadata(user, dataset.storageSourceId());
+    EntityTypeMetadata obj = ent.get(tableName);
+    return new DatasetPreviewTable()
+        .columns(convertEntityToColumn(obj))
+        .rows(
+            rawlsService
+                .entityQuery(user, dataset.storageSourceId(), tableName)
+                .getResults()
+                .stream()
+                .map(Entity -> convertEntityToRow(Entity, obj.getIdName()))
+                .toList());
+  }
+
+  private Object convertEntityToRow(Entity entity, String idName) {
+    @SuppressWarnings("unchecked")
+    Map<String, String> att = (Map<String, String>) entity.getAttributes();
```

Comment:
This should also be fixable by updating the Rawls swagger.

Reply:
Which will be done in a Rawls PR, not in this PR. If it's not in before this is merged, it's ok to keep the cast in for now.